### DISCUSSION TITLE
orderByFallback() function request

### DISCUSSION THREAD

**Discussion by kellyannmcnamara01** [craftcms/cms#11457] [Ideas] [answered]
## Concept
Sort query by the score with a fallback selected. The fallback orderBy would only apply if there were entries/users that have the same 'score'. This orderByFallback function would be applicable to entries and users and only used when the primary orderBy is 'score'.

### Example
Your website is an organization with a directory of staff that people can search through. After a user applies their selected filters the query orders by score. If there are three staff members that all have the same score, the ordering of said three staffers would fall back on the selected method; i.e type of employee (full-time, part-time, volunteer).

### Fallback Types Requested
- a select field (example: employee type; full-time, part-time, volunteer)
- user group
- title (or name for users)
- publish date (or join date for users)
- other appliable fields other users may want [/shrug]

_I particularly need it for the select field or user group_

**Comment by brandonkelly** [comment, marked as answer]
Great idea. I’ve just made this possible for Craft 4.1, all via existing `orderBy` param (#11470). Going forward you will be able to set that to `score` + additional column values, e.g.

```twig
{% set entries = craft.entries()
  .search('foo')
  .orderBy('score, title asc')
  .all() %}
```

**Reply by brandonkelly** [reply]
Craft 4.1 is out with that change ✨
